analyzer/runtime/runtime.go

```diff
@@ -220,15 +220,21 @@ func (m *processor) queueDbUpdates(batch *storage.QueryBatch, data *BlockData) {
 		var error_module string
 		var error_code uint32
 		var error_message *string
+		var error_message_raw *string
 		if transactionData.Error != nil {
 			error_module = transactionData.Error.Module
 			error_code = transactionData.Error.Code
 			error_message = transactionData.Error.Message
+			error_message_raw = transactionData.Error.RawMessage
 		}
 		if error_message != nil {
 			// Apparently the message does need to be valid UTF-8.
 			// In the rare case it's not, hex encode it.
 			// https://github.com/oasisprotocol/nexus/issues/439
+			if !storage.IsValidText(*error_message_raw) {
+				*error_message_raw = hex.EncodeToString([]byte(*error_message_raw))
+			}
+			// this is likely unnecessary. todo: remove
```

Comment:
Agreed that it can be removed below since the error_message string is sanitized via `sanitizeString`?

Actually, raw message is sanitized as well, so the above check is not needed either, not sure why it was here before even.

---

I did a minor investigation and I see what happened: 
- Mitja fixed this by sanitizing error messages in: https://github.com/oasisprotocol/nexus/pull/373 (deployed in v0.0.15).
- I fixed this by adding the `storage.IsValidText` check in: https://github.com/oasisprotocol/nexus/pull/440 - because staging at the time was running an older version that did not contain the Mitja's fix and I did not notice that this was already fixed.

So the checks here are redundant, feel free to remove the checks.

Reply:
removed! I updated the tests for `IsValidText` to now test `SanitizeString` as well